wrapper/hosts.py

```diff
@@ -364,9 +364,34 @@ def handle_finish(self, data, state):
                           state['internal']['disk_ids'])
             logging.debug('Assumed volume list: %r', volumes)
             return False
-        # Move volumes to destination project
         for vol in volumes:
-            logging.info('Transfering volume: %s', vol)
+            logging.info('Transferring volume: %s', vol)
+            # Checking if volume is in available state
+            is_available = False
+            start_at = time.time()
+            while start_at + TIMEOUT > time.time():
+                volume_state = self._run_openstack([
+                        'volume', 'show', '-f', 'value', '-c', 'status', vol,
+                        ], data)
+                if volume_state is None:
+                    error('Unable to get volume state, quitting.')
+                    return False
+                volume_state = volume_state.rstrip()
+                logging.info('Current volume state: %s.', volume_state)
+                if volume_state == 'available':
+                    logging.info(
+                        'Volume detached n first %s second(s), trasferring.',
```

Comment:
`... n first` probably should have been `... in first`

Reply:
You are right. This was a typo! 
While I was correcting it I saw another small issue which I changed as well. 
I changed the result into Implicitly declared _int_ because it defaulted to a bit ugly _float_.